openfisca_core/variables.py

```diff
@@ -187,6 +188,37 @@ def __init__(self, baseline_variable = None):
                 .format(self.name, ', '.join(sorted(unexpected_attrs.keys()))))
 
         self.is_neutralized = False
+        self._deps = []
+
+    # ----- properties ----- #
+    @property
+    def dependencies(self):
+        """
+        Returns the variables this variable depends on. For this to work
+        the formula must use constant strings as the first input to entity when
+        retrieving the variable.
+
+        Please be mindful that this function simply checks all the constant strings defined
+        in the formula, so avoid using a constant string that collides with a variable
+        name, as it'll lead to incorrect dependencies being reported.
+
+        Input variables return an empty list.
+        """
+        if self.is_input_variable() or self._deps:
+            return self._deps
+        code = self.get_formula().__code__
```

Comment:
@Morendil @eraviart @magemax Would you have some advice on how to inspect a Python formula content? 
We used the `inspect` module in other Web API code sections but I didn't find how to get information on the called "helpers"/dependencies.
> Here is [some documentation](https://docs.python.org/3.7/library/inspect.html) on the `inspect` module and `code` attributes.

Reply:
@sandcha @sarafalamaki At the risk of not answering the question directly, and being less helpful than I like, there's been a lot of previous discussion on the problems and limitations of parsing Python to figure out what variables a formula depends on; that was part of the point of my [proposal to reform formula syntax](https://github.com/openfisca/openfisca-core/issues/890) towards something more declarative.

This PR has a clever way of achieving the desired end result, using `co_consts` on the bytecode; it will do so with an accuracy that's difficult to estimate, but to hazard a guess I'd expect around 80-90% - this will be more than enough for relatively new country packages that don't suffer from the accumulated years of cruft of e.g. Openfisca-France, but it's bound to fail occasionally on a variety of edge cases and so I'd expect it to turn into a maintenance headache, to some degree.

My best advice would be not to do this and instead invest first in the proposed changes to formula syntax. I realize that this may be unpalatable advice, and I can understand if the absence of this feature is a deal-breaker for using Openfisca at all.

The second best course of action is to only do this if you're fully prepared to commit time and money to a roadmap for improving formula syntax such that statically computing formula dependencies can be done with 100% accuracy. In that case, yes, this PR could be seen as "prototyping the API contract for retrieving variable dependencies", with the option to then swap out the approximate bytecode-based implementation for an accurate, "architecture-based" one.

Be aware, such a choice comes with a serious risk - that of fooling oneself into thinking "we'll do the quick thing now and get around to cleaning up the technical debt later", and then, as one does, never getting around to cleaning up later. That is the very substance of technical debt.